internal/pkg/queue/actionqueue.go

```diff
@@ -11,38 +11,49 @@ import (
 	"github.com/elastic/elastic-agent/internal/pkg/fleetapi"
 )
 
+// persistor is an the minimal interface needed for state storage.
+type persistor interface {
+	SetQueue(a []fleetapi.Action)
+	Save() error
```

Comment:
[NIT]
If it's a persistor, it should have a persist method :)

Reply:
changed to `saver`